src/test/java/geocode/ReversePostalCodeTest.java

```diff
@@ -0,0 +1,47 @@
+package geocode;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.zip.ZipInputStream;
+
+public class ReversePostalCodeTest {
+
+    @Test
+    public void givenPostalCodeZipFile_constructor_buildsKdTree() throws IOException {
+        ZipInputStream fileStream = new ZipInputStream(Objects.requireNonNull(ClassLoader.getSystemResourceAsStream(
+                "US.zip")));
+        ReversePostalCode reversePostalCode = new ReversePostalCode(fileStream);
+        assertThat(reversePostalCode.nearestPostalCode(39.0955, -94.5844, null).postalCode,
+                is("64121"));
+    }
+
+    @Test
+    public void givenPostalCodeTextFile_constructor_buildsKdTree() throws IOException {
+        InputStream fileStream = ClassLoader.getSystemResourceAsStream("US.txt");
+        ReversePostalCode reversePostalCode = new ReversePostalCode(fileStream);
+        assertThat(reversePostalCode.nearestPostalCode(39.0955, -94.5844, null).postalCode,
+                is("64121"));
+    }
+
+    @Test
+    public void givenLocationInGreenlandWithMaxDistance2_findNearestPostalCode_shouldReturnNoPostalCode() throws IOException {
+        InputStream fileStream = ClassLoader.getSystemResourceAsStream("US.txt");
+        ReversePostalCode reversePostalCode = new ReversePostalCode(fileStream);
+        assertNull(reversePostalCode.nearestPostalCode(78.695697, -41.337372, 2D));
```

Comment:
I was going to suggest pretty much this exact test and then I scrolled down

Reply:
Should there also be a positive test to ensure that the `findNearestPostalCode` can return a non-null value if you give a non-null distance boundary?

Reply:
Yep, I'll add it.